### PR TITLE
QRL::bind() is QRL / Issue2972 rewritten

### PR DESCRIPTION
**Pull request by mesterum**
# Overview

# What is it?

- [ ] Feature / enhancement
- [x] Bug
- [ ] Docs / tests / types / typos

# Description

[PR for issue: 4223](https://github.com/BuilderIO/qwik/issues/4223)

# Checklist:

- [x] My code follows the [developer guidelines of this project](https://github.com/BuilderIO/qwik/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] Replaced the Issue2972 test with mine

### CODE REVIEW

**Comment by mesterum** [comment]
I admit that this solution is half baked. It doesn’t hold over the wire. The bind arguments must be captured too.

File: packages/qwik/src/core/qrl/qrl-class.ts [hunk 1]
```diff
@@ -159,7 +159,9 @@ export const createQRL = <TYPE>(
     $captureRef$: captureRef,
     dev: null,
   };
-  const qrl = Object.assign(invokeQRL, methods);
+  Object.setPrototypeOf(methods, Function.prototype)
+  seal(methods)
+  const qrl = Object.setPrototypeOf(invokeQRL, methods);
```
Comment (mhevery) [inline packages/qwik/src/core/qrl/qrl-class.ts:164]: Why do we have to do this? This has a big performance penalty.: https://stackoverflow.com/questions/52654781/why-does-using-object-setprototypeof-impact-performance-heavily

**Review by mhevery — CHANGES_REQUESTED** [review]
1. It introduces a negative performance of using `Object.setProtypeOf`
2. It will not survive serialization over the wire.

**Comment by mhevery** [comment]
If there is no progress on this PR in the next few weeks, I would suggest closing it.

**Comment by mesterum** [comment]
I didn’t know about the penalty of using Object.setPrototypeOf(). Using
Object.create() is not possible be instead of the last line because the qrl
must be a callable function.
Why do I want the methods be in the prototype chain(PC). Because the
f.bind(…args) function creates another function with all the PC of f. If
methods object is not in PC of qrl, it will be not in qrl.bind().

**Comment by mesterum** [comment]
1.
   2. It will not survive serialization over the wire.

I don’t know how to do it.

**Pull request closed without merging**